src/components/Profile/Profile.js

```diff
@@ -3,52 +3,60 @@ import './Profile.css';
 import { Col, Row } from 'react-bootstrap';
 import React, { Component } from 'react';
 
+import ReturnedComponents from '../../data/returned_components.json'
 import ReturningModal from './ReturningModal/ReturningModal';
 import { connect } from 'react-redux';
 
 class Profile extends Component {
     constructor(props) {
         super(props);
-        this.memberID = props.memberID ;
-        this.mock_reservations = props.mock_reservations;
+        /** @type { String } */
+        this.memberID = props.memberID;
+        /** @type { number } */
+        this.user_index_returned = ReturnedComponents.records.findIndex( 
+            reservation => reservation.memberID === this.memberID 
+        );
+        /** @type {!Array<{componentID: String, quantity: number, dateReturned: String}>, ...}>}*/
+        this.returned_components = (
```

Comment:
Maybe you can break this into a function? Something like 
`this.returned_components = getReturnedComponents(this.user_index_returned);`